core/src/main/java/org/mule/registry/TransientRegistry.java

```diff
@@ -283,7 +289,17 @@ public void registerObject(String key, Object object, Object metadata) throws Re
             return;
         }
 
-        registryMap.putAndLogWarningIfDuplicate(key, object);
+        doRegisterObject(key, object, metadata);
+    }
+
+    protected <T> T doGet(String key)
```

Comment:
There is no need to provide this method when the registryMap is already accesible